tests/test_default_router.py

```diff
@@ -727,3 +727,78 @@ def test_params_in_non_taken_branches(param_router, route, expected, num):
 
     assert resource.resource_id == num
     assert params == expected
+
+
+# capture path
+
+
+def test_capture_path_no_children():
+    router = DefaultRouter()
+    router.add_route('/foo/{bar:path}', ResourceWithId(1))
+    res = router.finder_src
+    with pytest.raises(
+        ValueError,
+        match='Cannot add route with template "/foo/{bar:path}/child". '
+        'Field name "bar" uses the converter "path"',
+    ):
+        router.add_route('/foo/{bar:path}/child', ResourceWithId(1))
+    with pytest.raises(
+        ValueError,
+        match='Cannot add route with template "/{bar:path}/child". '
+        'Field name "bar" uses the converter "path"',
+    ):
+        router.add_route('/{bar:path}/child', ResourceWithId(1))
+    assert res == router.finder_src
+
+
+@pytest.mark.parametrize(
+    'template',
+    (
+        '/foo/{bar:path}-x',
+        '/foo/x-{bar:path}',
+        '/foo/{x}-{bar:path}',
+        '/foo/{bar:path}-{x}',
+    ),
+)
+def test_capture_path_complex(template):
+    router = DefaultRouter()
+    with pytest.raises(
+        ValueError, match='Cannot use converter "path" of variable "bar" in a template '
+    ):
+        router.add_route(template, ResourceWithId(1))
+
+
+@pytest.fixture
+def capture_path_router():
+    router = DefaultRouter()
+
+    router.add_route('/foo/bar/baz', ResourceWithId(1))
+    router.add_route('/foo/{bar:path}', ResourceWithId(2))
+    router.add_route('/foo/bar/{foo:path}', ResourceWithId(3))
+    router.add_route('/{baz:path}', ResourceWithId(4))
+    router.add_route('/x/{v1:int}/{v2}/{other:path}', ResourceWithId(5))
+    return router
+
```

Comment:
We should also test that the following request paths result in 404s when `'/{baz:path}'` is *not* added as a route:

`'/foo'`
`'/foo/'`
`'/foo/bar'`
`'/foo/bar/'`